File: 3568-FindTheKeyOfTheNumbers/3568-FindTheKeyOfTheNumbers.py
# Last updated: 12/28/2025, 4:51:27 PM
class Solution:
    def generateKey(self, num1: int, num2: int, num3: int) -> int:
        num1_str = f"{num1:04}"
        num2_str = f"{num2:04}"
        num3_str = f"{num3:04}"
    
        key = ""
        for i in range(4):
            min_digit = min(num1_str[i], num2_str[i], num3_str[i])
            key += min_digit
        return int(key)

File: 3568-FindTheKeyOfTheNumbers/3568-FindTheKeyOfTheNumbers.java
// Last updated: 12/28/2025, 4:51:23 PM
class Solution {
    public int generateKey(int num1, int num2, int num3) {
        String num1Str = String.format("%04d", num1);
        String num2Str = String.format("%04d", num2);
        String num3Str = String.format("%04d", num3);
        StringBuilder key = new StringBuilder();
        for (int i = 0; i < 4; i++) {
            char minDigit = (char) Math.min(Math.min(num1Str.charAt(i), num2Str.charAt(i)), num3Str.charAt(i));
            key.append(minDigit);
        }
        return Integer.parseInt(key.toString());
    }
}